### DISCUSSION TITLE
Multi Column layout for dynamically parsed documents (One language on each side, of the same parsed document)

### DISCUSSION THREAD

**Discussion by christ-offer** [dolanmiu/docx#2330] [Q&A]
Hey. So I am currently working on a project where we want to generate word documents from some raw json. Ive made a parser, and it works perfectly fine as long as we don't want to have several languages (one language in each column for the same document).

For the working version with single column/single language, simply doing 
`children: parser(json)`, works fine.

But when trying to set this to two columns with a parsed result on each side. I'm running into issues.

Following the standard method with example text, renders fine:
```js
children: [
          new Paragraph({
              children: [
                  new TextRun("First Column"),
                  new ColumnBreak(),
                  new TextRun("Second Column"),
              ],
          }),
        ],
```

However trying to place the parser(json) instead of textruns/paragraphs/etc, fails.

```js
children: [
          new Paragraph({
              children: [
                  parser(json),
                  new ColumnBreak(),
                  parser(json),
              ],
          }),
        ],
```

And ditto with:

```js
children: [
          new Paragraph({
              children: [
                  new TextRun/Paragraph({children: parser(json)}),
                  new ColumnBreak(),
                  new TextRun/Paragraph(parser(json)),
              ],
          }),
        ],
```

This also does not work:

```js
children: [
            parser(json),
            new ColumnBreak(),
            parser(json),
        ],
```

The parsed array that is returned from parser(json) is an array full of new Paragraphs, new Tables, etc. It is returned atm as an any[].

Anyone have any ideas here? Do I have to rewrite the parser so that each new Paragraph/Table etc gets its own section?

**Comment by cesarmtzrdz** [comment]
Hello

Did you manage to make it work?
